disnake/message.py

```diff
@@ -153,7 +153,15 @@ async def _edit_handler(
     if embed is not MISSING:
         embeds = [embed] if embed else []
     if embeds is not MISSING:
-        payload["embeds"] = [e.to_dict() for e in embeds]
+        _embeds = []
+        for e in embeds:
+            if e._files:
+                for _file_key, file in list(e._files.items()):
+                    if file.closed:
+                        e._files.pop(_file_key)
+            _embeds.append(e.to_dict())
+
+        payload["embeds"] = _embeds
         for embed in embeds:
```

Comment:
This loops over `embeds` twice, could the two loops be merged into one?
(perhaps something like `files.extend(f for f in embed._files.values() if not f.closed)`? not sure.)

Reply:
Whit `files` are you referring to `_files` or `files`? If you're referring to `files` this won't work since if a user doesn't pass files `files` will be `MISSING` and not a List, also i'm not sure how they'll be handled by the lib, will they be send as embed image? 

Or I may not have understood what you mean.

Reply:
Wasn't clear enough, my bad.
I meant the last part here:
https://github.com/DisnakeDev/disnake/blob/46f769ab411473fe9d6a076f8d143989583a3dab/disnake/message.py#L156-L168
From what I can tell, the first loop could be merged into the second one, using the snippet mentioned in the first comment in place of line 168, or something similar.